File: tests/fixtures/agent_generated/agent_csrf_missing.py
# Agent-generated: AI assistant creates state-changing POST routes without CSRF
# Expected: SEC-009 (csrf_state_change)

from flask import Flask, jsonify, request

app = Flask(__name__)

@app.route("/api/transfer", methods=["POST"])
def transfer_funds():
    """Transfer money — no CSRF token validation."""
    data = request.json
    from_account = data.get("from")
    to_account = data.get("to")
    amount = data.get("amount")
    # Process transfer without CSRF protection
    return jsonify({"status": "transferred", "amount": amount})

@app.route("/api/settings", methods=["POST"])
def update_settings():
    """Update user settings — no CSRF protection."""
    settings = request.form.to_dict()
    return jsonify({"status": "updated"})

@app.route("/api/password", methods=["POST"])
def change_password():
    """Change password — no CSRF token check."""
    old_pw = request.form.get("old_password")
    new_pw = request.form.get("new_password")
    return jsonify({"status": "password changed"})

File: tests/fixtures/agent_generated/agent_csrf_missing.ts
// Agent-generated: AI assistant creates POST routes without CSRF protection
// Expected: SEC-009 (csrf_state_change_ts)

import express from "express";

const app = express();
app.use(express.json());

// No CSRF middleware applied

app.post("/api/transfer", (req, res) => {
  // AI generated: state-changing operation without CSRF token
  const { from, to, amount } = req.body;
  res.json({ status: "transferred", amount });
});

app.post("/api/settings", (req, res) => {
  // AI generated: settings update without CSRF protection
  const settings = req.body;
  res.json({ status: "updated", settings });
});

app.post("/api/password", (req, res) => {
  // AI generated: password change without CSRF token validation
  const { oldPassword, newPassword } = req.body;
  res.json({ status: "password changed" });
});

app.put("/api/profile", (req, res) => {
  // AI generated: profile update via PUT without CSRF
  res.json({ status: "profile updated" });
});
